test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c

```diff
@@ -6,10 +6,63 @@
 #include "FreeRTOS_IP.h"
 #include "FreeRTOS_IP_Private.h"
 #include "FreeRTOS_ARP.h"
+#include "FreeRTOS_Routing.h"
+
+/* Global variables accessible throughout the program. Used in adding
+ * Network interface/endpoint. */
+NetworkInterface_t xNetworkInterface1;
+NetworkEndPoint_t xEndPoint1;
+
+const uint8_t ucIPAddress2[ 4 ];
+const uint8_t ucNetMask2[ 4 ];
+const uint8_t ucGatewayAddress2[ 4 ];
+const uint8_t ucDNSServerAddress2[ 4 ];
+const uint8_t ucMACAddress[ 6 ];
 
 void harness()
 {
+    NetworkBufferDescriptor_t xLocalNetworkBufferDescriptor;
     ARPPacket_t xARPFrame;
+    NetworkEndPoint_t xLocalEndPoint;
+
+    /* The Ethernet buffer must contain the ARP packet. */
+    xLocalNetworkBufferDescriptor.pucEthernetBuffer = &xARPFrame;
+
+    /* Non-deterministically add an end-point to the descriptor. */
+    if( nondet_bool() )
+    {
+        xLocalNetworkBufferDescriptor.pxEndPoint = NULL;
```

Comment:
Can you use conditional assignment x = nondet_bool() ? NULL : expression

?

Reply:
Sure! that works for me :)